Add test for relative seconds, and rename conditions

matt-o-how opened the pull request:
This PR adds a new test for relative time, renames AGG_SIG to AGG_SIG_UNSAFE, and renames the time/height conditions to use "absolute" and "relative" naming conditions

matt-o-how committed:
Add test for relative seconds

matt-o-how committed:
Merge branch 'main' into time_conditions_changes

richardkiss commented:
Seems okay. I thought we'd talked about renaming the conditions `SECONDS_NOW` to `SECONDS_ABSOLUTE` and `SECONDS_AGE` to `SECONDS_RELATIVE`?

mariano54 commented on tests/blockchain/test_blockchain_transactions.py:
```diff
@@ -842,7 +842,67 @@ async def test_assert_block_age_exceeds(self, two_nodes):
+            if coin.puzzle_hash == coinbase_puzzlehash:
+                spend_coin_block_1 = coin
+
+        # This condition requires block1 coinbase to be spent after 30 seconds from now
```
Do you mean 300 seconds in the comment?

matt-o-how replied:
Changed to 300

mariano54 commented on tests/blockchain/test_blockchain_transactions.py:
```diff
@@ -842,7 +842,67 @@ async def test_assert_block_age_exceeds(self, two_nodes):
+            guarantee_transaction_block=True,
+        )
+
+        # Try to validate that block before 30 sec
```
300 or 30?

matt-o-how replied:
Fixed

matt-o-how committed:
rebased on top of main

matt-o-how committed:
Renamed AGG_SIG to AGG_SIG_UNSAFE
Renamed height and time conditions
Changed some demo p2 puzzles to use AGG_SIG_ME

matt-o-how committed:
Merge branch 'main' into time_conditions_changes

matt-o-how committed:
fix merge on cost_calculator

matt-o-how committed:
update hex and sha256tree of p2_delegated_conditions

matt-o-how committed:
fix condition name in rl clvm

matt-o-how committed:
updated the innerpuz sha256tree after careful inspection

richardkiss commented on chia/full_node/mempool_check_conditions.py:
```diff
@@ -65,11 +65,11 @@ def mempool_assert_block_age_exceeds(
         return Err.INVALID_CONDITION
     if prev_transaction_block_height < block_index_exceeds_this:
-        return Err.ASSERT_HEIGHT_AGE_EXCEEDS_FAILED
+        return Err.ASSERT_HEIGHT_RELATIVE_FAILED
```
Shouldn't this be `<=`?

richardkiss replied:
In the time-based ones, it's `<=`. And "exceeds" to me suggests "strictly exceeds" (ie. 4 does not exceed 4, but 5 does).

richardkiss commented on chia/full_node/mempool_check_conditions.py:
```diff
@@ -49,11 +49,11 @@ def mempool_assert_block_index_exceeds(
         return Err.INVALID_CONDITION
     if prev_transaction_block_height < block_index_exceeds_this:
-        return Err.ASSERT_HEIGHT_NOW_EXCEEDS_FAILED
+        return Err.ASSERT_HEIGHT_ABSOLUTE_FAILED
```
`<=` ?

richardkiss commented on chia/util/errors.py:
```diff
@@ -127,7 +127,7 @@ class Err(Enum):
     INVALID_CC_CHALLENGE = 103
     INVALID_PREFARM = 104
-    ASSERT_SECONDS_AGE_EXCEEDS_FAILED = 105
+    ASSERT_SECONDS_RELATIVE_FAILED = 105
```
The fact that this is not near the others is a tad embarrassing.

richardkiss commented on chia/wallet/puzzles/p2_m_of_n_delegate_direct.clvm:
```diff
@@ -54,7 +54,7 @@
     (defun add_aggsig (public_key_list hash delegated_puzzle solution)
         (if public_key_list
-            (c (list AGG_SIG (f public_key_list) hash)
+            (c (list AGG_SIG_UNSAFE (f public_key_list) hash)
```
Pretty sure we want `AGG_SIG_ME` here.

richardkiss commented on chia/wallet/puzzles/condition_codes.clvm:
```diff
@@ -29,10 +29,10 @@
     ; wall-clock time
-    (defconstant ASSERT_SECONDS_AGE_EXCEEDS 80)
-    (defconstant ASSERT_SECONDS_NOW_EXCEEDS 81)
+    (defconstant ASSERT_SECONDS_RELATIVE 80)
```
When I see `ASSERT_SECONDS_RELATIVE 500` it's a lot less clear than `ASSERT_SECONDS_RELATIVE_EXCEEDS 500`, so I'm not really crazy about the name changes dropping "exceeds". Now there's no verb.

richardkiss replied:
That's probably why it was `AGE` & `NOW` before, to keep the name from being too long without losing any semantics.

wjblanke commented:
merged with richards OK
